sdk/tables/Azure.Data.Tables/src/Extensions/TableEntityExtensions.cs

```diff
@@ -39,6 +39,11 @@ internal static class TableEntityExtensions
                     case byte[] _:
                         annotatedDictionary[prop.Name.ToOdataTypeString()] = TableConstants.Odata.EdmBinary;
                         break;
+                    case BinaryData _:
+                        annotatedDictionary[prop.Name.ToOdataTypeString()] = TableConstants.Odata.EdmBinary;
+                        // Get the bytes from the BinaryData so that it serializes properly.
+                        annotatedDictionary[prop.Name] = ((BinaryData)annotatedDictionary[prop.Name]).ToArray();
```

Comment:
Can we avoid a copy here?

Reply:
Yeah, we could if we make a change to the utfJsonReader extensions to support `BinaryData`. I can look into that.

Reply:
Ah, it's that serializer :)